src/main/java/org/truffleruby/core/range/RangeNodes.java

```diff
@@ -546,16 +548,18 @@ protected RubyLongRange longRange(RubyClass rubyClass, long begin, long end, boo
             return range;
         }
 
-        @Specialization(guards = { "rubyClass != getRangeClass() || (!isImplicitLong(begin) || !isImplicitLong(end))" })
+        @Specialization(guards = { "nonStandardClass || (!isImplicitLong(begin) || !isImplicitLong(end))" })
```

Comment:
This variable is called `nonStandardClass` class, and it's set by `!=`, and it's used with `!`. Maybe we should invert it as `standardClass`, then it could be `==` and without `!`.

Reply:
That's what I was thinking, all the negation was excessive